src/libraries/System.Private.CoreLib/src/System/Char.cs

```diff
@@ -225,6 +225,76 @@ public static bool TryParse([NotNullWhen(true)] string? s, out char result)
         //
         // Static Methods
         //
+
+        /// <summary>Indicates whether a character is categorized as an ASCII letter.</summary>
+        /// <param name="c">The character to evaluate.</param>
+        /// <returns>true if <paramref name="c"/> is an ASCII letter; otherwise, false.</returns>
+        /// <remarks>
+        /// This determines whether the character is in the range 'A' through 'Z', inclusive,
+        /// or 'a' through 'z', inclusive.
+        /// </remarks>
+        public static bool IsAsciiLetter(char c) => (uint)((c | 0x20) - 'a') <= 'z' - 'a';
```

Comment:
Open question: is `((uint)(c - 'A') & ~0x20u) < 26` a more effiicent pattern because it allows folding the MOV and SUB into a single LEA operation? (I don't know the answer to this re: perf, but it is a codegen size savings on x86.)

Reply:
Do you have a sharplab example that shows the lea?  I don't see a codegen size difference on x86, but I might be doing it wrong.

Reply:
```cs
using System;

foreach (Char ch in "Hello!")
{
    if (IsAsciiLetter1(ch)) Console.WriteLine(ch);
}

foreach (Char ch in "Hello!")
{
    if (IsAsciiLetter2(ch)) Console.WriteLine(ch);
}

static bool IsAsciiLetter1(char c) => (uint)((c | 0x20) - 'a') <= 'z' - 'a';
static bool IsAsciiLetter2(char c) => (uint)((c - 'A') & ~0x20) < 26;
```
```asm
; Core CLR 6.0.322.12309 on amd64

<Program>$.<Main>$(System.String[])
    L0000: push rdi
    L0001: push rsi
    L0002: push rbp
    L0003: push rbx
    L0004: sub rsp, 0x28
    L0008: mov rcx, 0x19f2a7d1c18
    L0012: mov rsi, [rcx]
    L0015: mov rdi, rsi
    L0018: xor ebx, ebx
    L001a: mov ebp, [rdi+8]
    L001d: test ebp, ebp
    L001f: jle short L0041
    L0021: movsxd rcx, ebx
    L0024: movzx ecx, word ptr [rdi+rcx*2+0xc]
    L0029: mov eax, ecx
    L002b: or eax, 0x20 ; <-- OR between MOV and ADD, preventing combination
    L002e: add eax, 0xffffff9f
    L0031: cmp eax, 0x19
    L0034: ja short L003b
    L0036: call 0x00007ffbe91e41d8
    L003b: inc ebx
    L003d: cmp ebp, ebx
    L003f: jg short L0021
    L0041: mov rdi, rsi
    L0044: xor ebx, ebx
    L0046: test ebp, ebp
    L0048: jle short L0068
    L004a: movsxd rcx, ebx
    L004d: movzx ecx, word ptr [rdi+rcx*2+0xc]
    L0052: lea eax, [rcx-0x41]
    L0055: and eax, 0xffffffdf ; <-- AND after MOV and ADD, allowing them to be combined into LEA
    L0058: cmp eax, 0x1a
    L005b: jae short L0062
    L005d: call 0x00007ffbe91e41d8
    L0062: inc ebx
    L0064: cmp ebp, ebx
    L0066: jg short L004a
    L0068: add rsp, 0x28
    L006c: pop rbx
    L006d: pop rbp
    L006e: pop rsi
    L006f: pop rdi
    L0070: ret

<Program>$.<<Main>$>g__IsAsciiLetter1|0_0(Char)
    L0000: movzx eax, cx
    L0003: or eax, 0x20
    L0006: add eax, 0xffffff9f
    L0009: cmp eax, 0x19
    L000c: setbe al
    L000f: movzx eax, al
    L0012: ret

<Program>$.<<Main>$>g__IsAsciiLetter2|0_1(Char)
    L0000: movzx eax, cx
    L0003: add eax, 0xffffffbf
    L0006: and eax, 0xffffffdf
    L0009: cmp eax, 0x1a
    L000c: setb al
    L000f: movzx eax, al
    L0012: ret
```